js/src/widgets/thumbnailsView.js

```diff
@@ -137,26 +139,33 @@
 
     loadImages: function() {
       var _this = this;
-      var delayTimes = this.state.getStateProperty('delayTimes');
-      var mainImageDelay = parseInt(delayTimes == undefined? 1000 : delayTimes.mainImage);
-      var thumbnailDelay = parseInt(delayTimes == undefined? 250 : delayTimes.thumbnails);
-      jQuery.each(_this.element.find("img"), function(key, value) {
-        setTimeout( function(){
+      if (!_this.parsedImageElements) {
+        jQuery.each(_this.element.find("img"), function(key, value) {
           if ($.isOnScreen(value, _this.lazyLoadingFactor) && !jQuery(value).attr("src")) {
             var url = jQuery(value).attr("data");
-            _this.loadImage(value, url);
+            _this.imagesToLoadList.push({"value": value, "url": url});
           }
-        }, mainImageDelay);
-        mainImageDelay += thumbnailDelay;
-      });
+        });
+        _this.imagesToLoadList.reverse();
+        _this.parsedImageElements = true;
+      }
+
+      if(_this.imagesToLoadList.length) {
```

Comment:
@siddharthbhagwan To help clarify the intent of your code, extract this `if/then/else` block into its own function, `loadNextImage`. Call it once here, and then, in the `loadImage` function, call `_this.loadNextImage()` instead of `_this.loadImages()`.